doc/source/user/how-to-index.rst

```diff
@@ -0,0 +1,324 @@
+    <BLANKLINE>
+            [[25, 26, 27, 28, 29]]])
+
+Index columns
+-------------
+
+Use :ref:`dimensional-indexing-tools` to avoid shape mismatches::
+
+    >>> arr = np.arange(3*4).reshape(3, 4)
+    >>> arr
+    array([[ 0,  1,  2,  3],
+           [ 4,  5,  6,  7],
+           [ 8,  9, 10, 11]])
+    >>> column_indices = [[1, 3], [0, 2], [2, 2]]
+    >>> np.arange(arr.shape[0])
+    array([0, 1, 2])
+    >>> row_indices = np.arange(arr.shape[0])[:, np.newaxis]
+    >>> row_indices
+    array([[0],
+           [1],
+           [2]])
+
+Use the ``row_indices`` and ``column_indices`` for advanced
+indexing::
+
+    >>> arr[row_indices, column_indices]
+    array([[ 1,  3],
+           [ 4,  6],
+           [10, 10]])
+
+Create subsets of larger matrices
+=================================
+
+Use :ref:`slicing-and-striding` to access chunks of a large array::
+
+    >>> a = np.arange(100).reshape(10, 10)
+    >>> a
+    array([[ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9],
+            [10, 11, 12, 13, 14, 15, 16, 17, 18, 19],
+            [20, 21, 22, 23, 24, 25, 26, 27, 28, 29],
+            [30, 31, 32, 33, 34, 35, 36, 37, 38, 39],
+            [40, 41, 42, 43, 44, 45, 46, 47, 48, 49],
+            [50, 51, 52, 53, 54, 55, 56, 57, 58, 59],
+            [60, 61, 62, 63, 64, 65, 66, 67, 68, 69],
+            [70, 71, 72, 73, 74, 75, 76, 77, 78, 79],
+            [80, 81, 82, 83, 84, 85, 86, 87, 88, 89],
+            [90, 91, 92, 93, 94, 95, 96, 97, 98, 99]])
+    >>> a[2:5, 2:5]
+    array([[22, 23, 24],
+           [32, 33, 34],
+           [42, 43, 44]])
+    >>> a[2:5, 1:3]
+    array([[21, 22],
+           [31, 32],
+           [41, 42]])
+    >>> a[:5, :5]
+    array([[ 0,  1,  2,  3,  4],
+           [10, 11, 12, 13, 14],
+           [20, 21, 22, 23, 24],
+           [30, 31, 32, 33, 34],
+           [40, 41, 42, 43, 44]])
+
+The same thing can be done with advanced indexing in a slightly complex
+way. Remember that advanced indexing creates a copy::
+
+    >>> a[np.arange(5)[:,None], np.arange(5)[None,:]]
+    array([[ 0,  1,  2,  3,  4],
+           [10, 11, 12, 13, 14],
+           [20, 21, 22, 23, 24],
+           [30, 31, 32, 33, 34],
+           [40, 41, 42, 43, 44]])
+
+You can also use :meth:`mgrid` to generate indices::
+
+    >>> indices = np.mgrid[0:6:2]
+    >>> indices
+    array([0, 2, 4])
+    >>> a[:, indices]
+    array([[ 0,  2,  4],
+           [10, 12, 14],
+           [20, 22, 24],
+           [30, 32, 34],
+           [40, 42, 44],
+           [50, 52, 54],
+           [60, 62, 64],
+           [70, 72, 74],
+           [80, 82, 84],
+           [90, 92, 94]])
+
+Filter values
+=============
+
+Non-zero elements
+-----------------
+
+Use :meth:`nonzero` to get a tuple of array indices of non-zero elements 
+corresponding to every dimension::
+
+	>>> z = np.eye(3)
+	>>> z
+	array([[1., 0., 0.],
+           [0., 1., 0.],
+           [0., 0., 1.]])
+	>>> np.nonzero(z)
+	(array([0, 1, 2]), array([0, 1, 2]))
```

Comment:
Can we change this to a slightly less trivial example? I think mixing values for indices should be avoided when possible